### ISSUE TITLE
Content that is Cast to TV is not captured

### ISSUE THREAD

**Issue by botagas** [TBSniller/piccap#79]
Hello,

I have been trying to find someone who has faced anything similar, searched all over to find at least a mention, but sadly - nothing.
What I am currently trying to overcome is the fact that streaming from phone/PC via AirPlay on my LG C2 doesn't work with capturing.

My current setup is a Hyperion instance running on Debian 11 in my server. 
The TV is rooted, has PicCap installed and successfully captures content when watching YouTube, watching movies / series via Jellyfin or watching content on Kodi or Browser.
I can also use a macbook to connect to the TV as a separate display via AirPlay. That works - the lights react to what happens on the screen. Same goes for my Windows PC.

When I checked the Live Video from the capture, content that is streamed via AirPlay is not visible - the capture sees a black screen. The player UI itself is visible, but the video content is black. That happens with a simple YouTube video stream from phone/computer. It works fine when I just use the YouTube app instead of streaming it from an external device via AirPlay. 

Am I missing something or misunderstanding something? Is this a limitation of what the TV is able to capture?

TL;DR:
Streamed content via AirPlay appears a a black video in the capture.

**Comment by Dammouz** [comment]
Hello,
So you confirm that capture sample in Hyperion (ILed  Instances => Led Ouput => Led Layout tab => Live Video) shows also black sreen. And when the video is inside an app window, and you move this window, is it only the content that is black ?
If so, it is due to the DRM. AirPlay, like Chrome & Firefox, blocks some content to be captured. For example, on Windows, It is black for Netflix & Disnay+ when viewing it through web browser.
This is not related to Hyperion nor PicCap. But it is to "protect" the media against copying.

**Comment by botagas** [comment]
> Hello, So you confirm that capture sample in Hyperion (ILed Instances => Led Ouput => Led Layout tab => Live Video) shows also black sreen. And when the video is inside an app window, and you move this window, is it only the content that is black ? If so, it is due to the DRM. AirPlay, like Chrome & Firefox, blocks some content to be captured. For example, on Windows, It is black for Netflix & Disnay+ when viewing it through web browser. This is not related to Hyperion nor PicCap. But it is to "protect" the media against copying.

Sorry for the insanely late reply as I have cast aside this due to limited time.
As it is currently, ANY content that's cast over AirPlay is displayed as black in capture sample (Live Video). However, if you use Screen Mirroring (Mac to TV), the video feed is fine. 

Even streaming a YouTube video from iPhone/Macbook outputs a black live video. However, if I used Screen Mirroring instead, it doesn't matter if the content is DRM-protected or not, I still get a perfectly fine Live Video feed. However, it isn't very convenient that I have to use Screen Mirroring any time I want to use the ambient lightning for content that I want to stream to the TV.

**Comment by TBSniller** [comment]
@botagas I've just tried to play AirPlay content from an iOS device on my WebOS5 TV. It was displayed and captured without any problems or hints on DRM content. 

From what you describe it really sounds like the DRM content protection is hitting. You can verify it by checking the logs of hyperion-webos. 

For that go into PicCap's `log`-tab before you play the video and hit the `setup logging` button. After that you can hit the `restart service` button. You now may have to re`start` your capture on the `first tab` to trigger service startup. 
After that you can go back to `log`-tab and hit the `hyperion-webos` button. In here you can `load the last 200 log lines` to verify logging is working.
Now replicate your issue by playing AirPlay content and getting your black capture output. Then go back into PicCaps `log`-tab, open `hyperion-webos` logs with the button and `load last 200 log lines`. 

You may now see some error messages with `error 17`, which means that you encounter DRM protected content. In this case we can do nothing due to https://github.com/TBSniller/piccap/issues/38. If you receive another error regarding the capture process, feel free to open an issue here in the service repository https://github.com/webosbrew/hyperion-webos.
